### PR TITLE
receiver/prometheus: add metricGroup.toNumberDataPoint pdata conversion

### PR DESCRIPTION
**Pull request by odeke-em**
Implements metricGroupPdata toNumberDataPoint and added unit tests
as well as equivalence tests to ensure the migration will render
the same results.

Updates #3137
Depends on PR #3668
Updates PR #3427

### CODE REVIEW

**Comment by alolita** [comment]
@dashpole @Aneurysm9 @rakyll please review.

**Review by dashpole — APPROVED** [review]
toNumberDataPoint implementation and testing looks good.

**Comment by punya** [comment]
@odeke-em could you please appease the linter:

```
receiver/prometheusreceiver/internal/otlp_metricfamily_test.go:509: File is not `gofmt`-ed with `-s` (gofmt)
        type scrape struct {
```

**Comment by tigrannajaryan** [comment]
@Aneurysm9 @rakyll do you plan to review? If not I can merge based on @dashpole's approval.

**Comment by odeke-em** [comment]
Thanks for the reviews @dashpole @tigrannajaryan! @bogdandrutu this one too is ready :-)

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go [hunk 1]
```diff
@@ -172,6 +172,23 @@ func (mg *metricGroupPdata) toSummaryPoint(orderedLabelKeys []string, dest *pdat
 	return true
 }
 
+func (mg *metricGroupPdata) toNumberDataPoint(orderedLabelKeys []string, dest *pdata.NumberDataPointSlice) bool {
+	var startTsNanos pdata.Timestamp
+	tsNanos := pdata.Timestamp(mg.ts * 1e6)
+	// gauge/undefined types have no start time.
+	if mg.family.isCumulativeTypePdata() {
+		startTsNanos = tsNanos
```
Comment (bogdandrutu) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:180]: This is not correct. The starttime should not be set to the current timestamp, but to the beginning of the interval.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:180]: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L132

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:182]: Got it, thank you @bogdandrutu! That means that the prior code was incorrect as this is a translation. I have filed this issue https://github.com/open-telemetry/opentelemetry-collector/issues/3691 as a bug

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:182]: @bogdandrutu please take a look, I've filed the appropriate bug and I've added TODOs to ensure that after this merge I shall send a follow up to fix that problem. Thank you!

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:182]: @bogdandrutu I've mailed out PR https://github.com/open-telemetry/opentelemetry-collector/pull/3694 which is based off this one, and after this one is merged, we'll rebase. Thank you

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**